src/coreclr/src/vm/diagnosticsprotocol.h

```diff
@@ -47,11 +47,20 @@ bool TryParseString(uint8_t *&bufferCursor, uint32_t &bufferLen, const T *&resul
     }
     if (stringLen > (bufferLen / sizeof(T)))
         return false;
-    if ((reinterpret_cast<const T *>(bufferCursor))[stringLen - 1] != 0)
+
+    T terminator;
+    memcpy(&terminator, bufferCursor + stringLen - 1, sizeof(T));
+    if (terminator != 0)
         return false;
-    result = reinterpret_cast<const T *>(bufferCursor);
+
+    T *resultBuffer = new (nothrow) T[stringLen];
```

Comment:
I just updated the code so that strings are held in `NewHolder|NewArrayHolder`s so they should get deleted correctly.  I'm going to see about running Valgrind over it to make sure.